src/SFML/Network/Packet.cpp

```diff
@@ -571,6 +571,8 @@ Packet& Packet::operator<<(const String& data)
 ////////////////////////////////////////////////////////////
 bool Packet::checkSize(std::size_t size)
 {
+    // Unsigned overflow is defined behavior
+    m_isValid = m_isValid && !(m_readPos + size < m_readPos);
     m_isValid = m_isValid && (m_readPos + size <= m_data.size());
```

Comment:
```suggestion
    // Determine if size is big enough to trigger an overflow
    const bool overflowDetected = (m_readPos + size < m_readPos);
    m_isValid = m_isValid && (m_readPos + size <= m_data.size()) && ! overflowDetected
```
Let's put a name to this new expression we're adding. Otherwise I like this and we can get this merged!